service/history/historybuilder/history_event_factory.go

```diff
@@ -36,138 +36,15 @@ import (
-		DBBufferBatch []*historypb.HistoryEvent
-		// whether to clear buffer events on DB
-		DBClearBuffer bool
-		// accumulated buffered events, equal to all buffer events from execution table
-		MemBufferBatch []*historypb.HistoryEvent
-		// scheduled to started event ID mapping for flushed buffered event
-		ScheduledIDToStartedID map[int64]int64
-	}
-
-	TaskIDGenerator func(number int) ([]int64, error)
-
-	BufferedEventFilter func(*historypb.HistoryEvent) bool
-
-	HistoryBuilder struct {
-		state           HistoryBuilderState
-		timeSource      clock.TimeSource
-		taskIDGenerator TaskIDGenerator
-
-		version     int64
-		nextEventID int64
-
-		// workflow finished
-		workflowFinished bool
-
-		// buffer events in DB
-		dbBufferBatch []*historypb.HistoryEvent
-		dbClearBuffer bool
-
-		// in mem events
-		memEventsBatches [][]*historypb.HistoryEvent
-		memLatestBatch   []*historypb.HistoryEvent
-		memBufferBatch   []*historypb.HistoryEvent
-
-		// scheduled to started event ID mapping
-		scheduledIDToStartedID map[int64]int64
-
-		metricsHandler metrics.Handler
-	}
-)
-
-func NewMutableHistoryBuilder(
-	timeSource clock.TimeSource,
-	taskIDGenerator TaskIDGenerator,
-	version int64,
-	nextEventID int64,
-	dbBufferBatch []*historypb.HistoryEvent,
-	metricsHandler metrics.Handler,
-) *HistoryBuilder {
-	return &HistoryBuilder{
-		state:           HistoryBuilderStateMutable,
-		timeSource:      timeSource,
-		taskIDGenerator: taskIDGenerator,
-
-		version:     version,
-		nextEventID: nextEventID,
-
-		workflowFinished: false,
-
-		dbBufferBatch:          dbBufferBatch,
-		dbClearBuffer:          false,
-		memEventsBatches:       nil,
-		memLatestBatch:         nil,
-		memBufferBatch:         nil,
-		scheduledIDToStartedID: make(map[int64]int64),
-
-		metricsHandler: metricsHandler,
-	}
-}
-
-func NewImmutableHistoryBuilder(
-	histories ...[]*historypb.HistoryEvent,
-) *HistoryBuilder {
-	lastHistory := histories[len(histories)-1]
-	lastEvent := lastHistory[len(lastHistory)-1]
-	return &HistoryBuilder{
-		state:           HistoryBuilderStateImmutable,
-		timeSource:      nil,
-		taskIDGenerator: nil,
-
-		version:     lastEvent.GetVersion(),
-		nextEventID: lastEvent.GetEventId() + 1,
-
-		workflowFinished: false,
-
-		dbBufferBatch:          nil,
-		dbClearBuffer:          false,
-		memEventsBatches:       histories,
-		memLatestBatch:         nil,
-		memBufferBatch:         nil,
-		scheduledIDToStartedID: nil,
-
-		metricsHandler: nil,
-	}
-}
-
-func (b *HistoryBuilder) IsDirty() bool {
-	return len(b.memEventsBatches) > 0 ||
-		len(b.memLatestBatch) > 0 ||
-		len(b.memBufferBatch) > 0 ||
-		len(b.scheduledIDToStartedID) > 0
-}
-
-// NOTE:
-// originalRunID is the runID when the WorkflowExecutionStarted event is written
-// firstRunID is the very first runID along the chain of ContinueAsNew and Reset
-
-func (b *HistoryBuilder) AddWorkflowExecutionStartedEvent(
+func (b *EventFactory) BuildWorkflowExecutionStartedEvent(
```

Comment:
Factory is a set of constructors. Constructors in go starts with `New`. I would rename all these factory methods to `NewXXX` instead of `BuildXXX`.